File: readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <string.h>
#include <math.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);
int coleman_liau(int letters, int words, int sentences);

int main(void)
{
    //TODO: Get string
    string text = get_string("Text: ");

    //TODO: count letters, words and sentences
    int letters = count_letters(text);
    int words = count_words(text);
    int sentences = count_sentences(text);

    //TODO: Coleman-liau count
    int col = coleman_liau(letters, words, sentences);

    //TODO: printf time
    if (col < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (col >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", col);
    }

}


int count_letters(string text)
{
    //TODO: Count numbers of letters
    int count = 0;
    for (int i = 0, len = strlen(text);  i < len; i++)
    {
        if (isalpha(text[i]))
        {
            count++;
        }
    }
    return count;
}

int count_words(string text)
{
    //TODO: Count numbers of words
    int count = 0;
    for (int i = 0, len = strlen(text); i < len; i++)
    {
        if (text[i] == ' ')
        {
            count++;
        }
    }
    return count + 1;
}

int count_sentences(string text)
{
    //TODO: Count numbers of sentences
    int count = 0;
    for (int i = 0, len = strlen(text); i < len; i++)
    {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            count++;
        }
    }
    return count;
}

int coleman_liau(int letters, int words, int sentences)
{
    //TODO: Count Coleman-liau
    float letter = (float) letters / words * 100;
    float sentence = (float) sentences / words * 100;

    int col = round(0.0588 * letter - 0.296 * sentence - 15.8);
    return col;

}

File: sentimental-readability/readability.py
# TODO:Here is the main function of the program.
def main():
    text = str(input("Text: "))
    letters = c_letters(text)
    words = c_words(text)
    sentences = c_sentences(text)
    liau = c_liau(letters, words, sentences)

    if liau < 1:
        print(f"Before Grade 1")
    elif liau >= 16:
        print(f"Grade 16+")
    else:
        print(f"Grade {liau}")


# TODO: This function counts the number of letters in the
def c_letters(text):
    count = 0
    for i in text:
        if i.isalpha():  # Test if the character is alphabetic
            count += 1
    return count


# TODO: This function counts the number of words in the text
def c_words(text):
    count = 0
    for i in text:
        if i.isspace():  # Test if the character is a white space
            count += 1
    return count + 1


# TODO: This function counts the number of sentences in the text
def c_sentences(text):
    count = 0
    for i in text:
        if i in ['.', '!', '?']:  # Test if the character is between ., ! or ?
            count += 1
    return count


# TODO: This function uses the Coleman-Liau formula to calculate the approximate reading level needed to understand the text
def c_liau(letters, words, sentences):
    l = ((letters / words) * 100)
    s = ((sentences / words) * 100)
    return round(0.0588 * l - 0.296 * s - 15.8)  # Coleman-Liau calculation


main()